File: OOP_Ponavljanje/c++/dodavanjeProfesora.cpp
#include<iostream>
using namespace std;

class Profesor {
public:
    string ime;
    string kolegiji[5];
    
    Profesor(string ime, string kolegiji[]) {
        this->ime = ime;
        int iterator = 0;
        while(!kolegiji[iterator].empty()) {
            this->kolegiji[iterator] = kolegiji[iterator];
            iterator++;
        }
        
    }
    
    
    
    void ispisKolegija() {
        int iterator = 0;
        while(!this->kolegiji[iterator].empty()) {
            cout << "Predaje: " << this->kolegiji[iterator] << endl;
            iterator++;
        }
    }
    
};

class Student {
    int matBr;
    static int numObj;
public:
    
    string ime;
    Student(string ime, int matBr) {
        Student::numObj++;
        this->ime = ime;
        this->matBr = matBr;
    }
    
    static int getNumObj() {
        return Student::numObj;
    }
    
};

int Student::numObj = 0;

int main(int argc, char** argv) {
    Student* pero = new Student("Petar", 1234);
    Student* mate = new Student("Mate", 3432);
    Profesor* profa = new Profesor("Miljenko", new string[5]{"PIN","DIMT","RIP"});
    profa->ispisKolegija();
    return 0;
}

File: OOP_Ponavljanje/php/dodavanjeProfesora.php
<?php 

class Profesor {
	public $ime;
	public $kolegiji = array();

	public function __construct($ime, array $kolegiji) {
		$this->ime = $ime;
		$this->kolegiji = $kolegiji;
		echo "Napravljen profa!";
	}

	public function ispisiKolegije() {
		foreach ($this->kolegiji as $kolegij)
			echo $kolegij . "<br/>";
	}

}

//definicija klase Student
class Student {
//podatkovni clanovi
	public $ime;
	private $matBr;

//konstruktor
	public function __construct($ime, $matBr) {
		$this->ime = $ime;
		$this->matBr = $matBr;
	}

//javna metoda
	public function getMatBr() {
		return $this->matBr;
	}
}

// $pero = new Student("Petar", 25);
// $ivana = new Student("Ivana", 123123);
// $mate = new Student("Mate", 112233);


// $profa = new Profesor("Miljenko", array('PIN', 'DIMT', 'RIP'));
// $profa->ispisiKolegije();